source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACRInputLabelView.mm

```diff
@@ -117,26 +116,52 @@ - (instancetype)initInputLabelView:(ACRView *)rootView acoConfig:(ACOHostConfig
             self.inputAccessibilityItem = accessibilityItem;
             self.inputAccessibilityItem.accessibilityLabel = inputView.accessibilityLabel;
         }
-
+        
         self.inputAccessibilityItem.isAccessibilityElement = YES;
         self.labelText = self.label.text;
-
+        
         if (HeightType::Stretch == inputBlck->GetHeight() && [inputView isKindOfClass:[ACRQuickReplyView class]]) {
             [self.stack addArrangedSubview:[(ACRColumnView *)viewGroup addPaddingFor:self]];
         }
-
+        
         self.shouldGroupAccessibilityChildren = NO;
-
+        
         NSObject<ACRIBaseInputHandler> *inputHandler = [self getInputHandler];
         inputHandler.isRequired = self.isRequired;
         inputHandler.hasValidationProperties |= inputHandler.isRequired;
         if (inputHandler.hasValidationProperties && errorMessage.empty()) {
             [rootView addWarnings:ACRMissingInputErrorMessage mesage:@"The input has validation, but there is no associated error message, consider adding error message to the input"];
         }
-
+        
         if (self.isRequired && (!self.label || !self.label.text.length)) {
             [rootView addWarnings:ACRMissingInputErrorMessage mesage:@"There exist required input, but there is no associated label with it, consider adding label to the input"];
         }
+        if(inputBlck->GetValueChangedAction() != nil ){
+            NSMutableArray *targetInputIds = [NSMutableArray array];
+            for (std::string &targetInputId : inputBlck->GetValueChangedAction()->GetTargetInputIds()) {
+                [targetInputIds addObject:[NSString stringWithCString:targetInputId.c_str() encoding:NSUTF8StringEncoding]];
+            }
+            if(targetInputIds != nil) {
+                self.targetInputIds = [NSArray arrayWithArray:targetInputIds];
+                [inputHandler addObserverWithCompletion:^{
+                    NSArray<ACRIBaseInputHandler> *_allInputHandlers = [rootView.card getInputs];
+                    for (NSObject<ACRIBaseInputHandler> *input in _allInputHandlers) {
+                        ACRInputLabelView *labelView = (ACRInputLabelView *)input;
+                        if (labelView) {
+                            if([self.targetInputIds containsObject:labelView.getInputHandler.id]) {
```

Comment:
I think targetInputIds can be kept as block variable in this method only, do we need instance variable?

Reply:
removed instance variable.